course-content/13_working_with_code.qmd

```diff
@@ -0,0 +1,397 @@
+
+
+<center>
+    <img src="images/working-with-code/google_syntax_error.png" size=500 style="margin:auto" border=1/>
+    <p style="text-align: center">
+        Error deep dive 
+    </p>
+</center>
+
+The highlighted summary information shows us that it understands that we are referring to a python issue, not a general coding one. So that is a good sign. It summarizes from a page on the realpython.com site (https://realpython.com/invalid-syntax-python/). Moreover it suggests different ways that others have extended the search that you might find helpful. While the suggested site itself is not an official python site, it nonetheless is a quality site about python that will likely be worthwhile to review. Once there, you will find more useful information 
+
+
+<br>
+
+
+:::{.callout-warning icon="false"}
+## Challenge 3
+
+You want to find out what the `print()` function does. How would you look to answer this question? What does it do, does it send a print job to a printer? 
+:::
+
+:::{.callout-note icon="false" collapse="true"}
+## Solution to Challenge 3
+Use one of the two most efficient ways to obtain information
+
+help(print)   # One way to get the info
+
+print() # Another way to get the info
+
+Prints the values passed into it to a stream or to an output (usually to output shown on a monitor)
+:::
+
+
+
+
+:::{.callout-warning icon="false"}
+## Challenge 4
+
+Someone asks you want the term "iterable" means in the context of a python function? How would you go about finding an answer? What does it mean? 
+
+:::
+:::{.callout-note icon="false" collapse="true"}
+## Solution to Challenge 4
+You could use help off of the menu. Might be quicker to do a Google search. 
+
+*Iterable is an object which can be looped over or iterated over with the help of a for loop. Objects like lists, tuples, sets, dictionaries, strings, etc. are called iterables. In short and simpler terms, iterable is anything that you can loop over.*
+
+You may or may not know what "loops" are, but perhaps have heard of lists and such. Particularly at the beginning of your Python journey, there will be terms that you will not yet be familiar with that will show up in explanations of other things. Don't worry! As with anything, this will occur less and less often over time. 
+:::
+
+<br>
+
+## Coding in style
+
+Working with python can be challenging. When you are first learning it, there may seem to be so many different rules around using certain kinds of brackets (and not others) at certain times, rules around indenting lines, etc. Many of these rules result in your code refusing to run. Often your code may run, however, it may not necessarily be easy to follow for someone reading through it. In an enterprise environment, your code will likely be read and referenced by others, so making it easy to decipher is important! 
```

Comment:
```suggestion
Working with python can be challenging. When you are first learning it, there may seem to be so many different rules around using certain kinds of brackets (and not others) at certain times, rules around indenting lines, etc. Many of these rules result in your code refusing to run. Often your code may run. However, it may not necessarily be easy to follow for someone else reading through it. In an enterprise environment, your code will likely be read and referenced by others, so making it easy to decipher is important! 
```